loyalty_criteria_multi_product/migrations/pre-migration.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def move_coupon_criteria_to_rule(env):
+    """Move everything related to coupon.criteria from the loyalty.program model to
+    loyalty.rule."""
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE loyalty_rule
+        ADD COLUMN IF NOT EXISTS coupon_criteria CHAR
+        """,
+    )
+    # If there are rules defined for the program, the value of coupon_criteria shall be
+    # defined in those rules, in case there are no previous rules for that program, a
+    # record shall be created for a new related rule to define coupon_criteria.
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE loyalty_rule AS lr
+        SET loyalty_criteria = 'multi_product'
+        FROM loyalty_program AS lp
+        WHERE lr.program_id = lp.id
+        AND lp.coupon_criteria = 'multi_product'
+        """,
+    )
```

Comment:
Shouldn't this be done after you insert the rules?

Reply:
You are right, thanks